File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {        
        Map<String, Integer> map = new HashMap<>();
        
        for(String[] clothe : clothes) {
            int count = map.getOrDefault(clothe[1], 0);
            map.put(clothe[1], count + 1);
        }
        
        if(map.size() == 1) {
            String[] cl = clothes[0];
            return map.get(cl[1]);
        } else {
            int an = 1;
            for(String key : map.keySet()) {
                an *= (map.get(key) + 1);
            }
            return an - 1;
        }
        
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
import collections;

def solution(clothes):
    dict = collections.defaultdict(int)
    answer = 1
    for cloth in clothes:
        dict[cloth[1]] += 1
    
    if len(dict) == 1:
        answer = dict[clothes[0][1]]
    else:        
        for key in dict:
            answer *= (dict[key] + 1)
        answer -= 1
        
    return answer
